### PR TITLE
[api][SIMS #611]Institution Admin Location Authorization

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Load all locations IDs when an Institution Admin is authenticated;
- Decorator @HasLocationAccess is now validating if Institution Admins have access to his locations only;
- Removed non longer necessary authorizations checks;
- Created unit tests for institution-location.guard;
- Created test helpers to mock the HTTP context.

### CODE REVIEW

File: sources/packages/api/src/services/institution-location/institution-location.service.ts [hunk 1]
```diff
@@ -144,4 +140,23 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
...
+   * Gets all locations ids for a particular institution.
+   * This method is used during the login process and should be
+   * as lightweight as possible. Do not expand this query unless
+   * it is related to login/authorization process.
+   * @param institutionId institution id.
+   * @returns institution locations ids.
+   */
+  async getInstitutionLocationsIds(institutionId: number): Promise<number[]> {
+    const allLocations = await this.repo
+      .createQueryBuilder("locations")
```
Comment (ann-aot) [inline sources/packages/api/src/services/institution-location/institution-location.service.ts:154]: table name is `institution_location` right? am I missing something?

Comment (andrewsignori-aot) [reply sources/packages/api/src/services/institution-location/institution-location.service.ts:154]: This is not a table name, it is the alias to execute the query. IMO using the full table name as an alias doesn't make sense and it also makes the code longer and hard to read.
Please let me know if it makes sense.

**Review by ann-aot — APPROVED** [review]
LGTM 👍

**Review by guru-aot — APPROVED** [review]
LGTM, good job on the unit test cases

**Review by dheepak-aot — APPROVED** [review]
LGTM

**Review by abschwenker — APPROVED** [review]

**Pull request merged**